schema/0007_add_parent_stop_id_to_historic_data.sql

```diff
@@ -0,0 +1,23 @@
+CREATE INDEX stops_parent_station_index
+    ON stops USING HASH (parent_station);
+
+CREATE INDEX stops_stop_id_index
+    ON stops USING HASH (stop_id);
+
+CREATE INDEX stops_vrr_timetable_version_index
+    ON stops USING HASH (vrr_timetable_version_id);
+
+ALTER TABLE historic_data ADD COLUMN parent_stop_id TEXT DEFAULT NULL;
+
+CREATE INDEX historic_data_parent_stop_id_index
+    ON historic_data USING HASH (parent_stop_id);
+
+UPDATE historic_data SET parent_stop_id = (
+    SELECT parent_stop_id FROM stops
+                          WHERE stops.stop_id = historic_data.stop_id
```

Comment:
- [ ] There seems to be some data, that is not matching up with stop_ids in stops table. At least in my local data there is for example "de:05954:5369" in delay data, but no direct match in stops table because in stops table it has "_Parent" suffix -> "de:05954:5369_Parent".

Reply:
After setting up some minimal local DB I´ve tested the updates. This UPDATE query produced `null` values only in my case. Also not sure if it aims to set the correct parents (formatting-wise) we are searching for here.

I´ve made some test and think following two UPDATEs are neccessary:

1)
```
UPDATE historic_data
SET parent_stop_id =
        (SELECT "NVBW_HST_DHID"
         FROM stops
         WHERE stops.stop_id = historic_data.stop_id
           AND stops.vrr_timetable_version_id = historic_data.vrr_timetable_version_id
         LIMIT 1)
WHERE parent_stop_id IS NULL;
```
2) 
As there are some rare cases where parent stop_id is already the one written in historic data, we just have to determine these entries and just "copy" the value over to parent_stop_id.
```
UPDATE historic_data
SET parent_stop_id =
        (select "NVBW_HST_DHID"
         FROM stops
         WHERE stops."NVBW_HST_DHID" = historic_data.stop_id
         LIMIT 1)
WHERE parent_stop_id IS NULL;
```
Alternatively you could apply a regex matching with parent ids (but who knows if there are some exceptions):
```
UPDATE historic_data
SET parent_stop_id = stop_id
WHERE parent_stop_id IS NULL
  AND stop_id ~ '^de:\d*:\d*$';
```

Reply:
I updated the crawler and I would we suggest we look tomorrow together at this.